File: julia/Comparatist/src/gcm/vec.jl
using ..Helper: init


function simulate!(x, r, epsilon)
    dim, steps = size(x)
    for t = 2:steps
        y = (1 - epsilon) * x[:, t-1] + epsilon * mean(x[:, t-1])
        x[:, t] .= r .* y .* (1 .- y)
    end
end


function prepare(name)
    res = init(name)
    function run()
        simulate!(res[:x], res[:r], res[:epsilon])
        res
    end
end

File: python/comparatist/gcm/vec.py
from ._helper import make_prepare


def simulate(x, f, r, epsilon):
    for i in range(1, len(x)):
        x[i] = f((1 - epsilon) * x[i-1] + epsilon * x[i-1].mean(), r)
    return x


prepare = make_prepare(simulate)
